src/Simplify.elm

```diff
@@ -11533,6 +11546,96 @@ setChecks collection =
         ]
 
 
+{-| Checks that when the first arguments of consecutive calls of the same function are equal,
+the effect of the earlier call gets "overwritten". For example
+
+    array |> Array.set 1 0 |> Array.set 1 100
+    --> array |> Array.set 1 100
+
+Use `operationDoesNotChangeResultOfOperationCheck`
+when _all arguments_ need to be the same, even if there is just one argument
+
+-}
+operationMakesPreviousOperationWithEqualFirstArgUnnecessaryCheck : { firstArgDescription : String } -> IntoFnCheck
+operationMakesPreviousOperationWithEqualFirstArgUnnecessaryCheck config =
```

Comment:
Alternative name suggestion that I find slightly easier to read, but choose what you prefer: `operationOverridesPreviousOperationWithEqualFirstArg`